demos/python_demos/im2latex_demo/im2latex_demo.py

```diff
@@ -0,0 +1,227 @@
+                      "in the console.",
+                      type=str)
+    args.add_argument("--vocab_path", help="Required. Path to vocab file to construct meaningful phrase",
+                      type=str, required=True)
+    args.add_argument("--max_formula_len",
+                      help="Optional. Defines maximum length of the formula (number of tokens to decode)",
+                      default="128", type=int)
+    args.add_argument("-d", "--device",
+                      help="Optional. Specify the target device to infer on; CPU, GPU, FPGA, HDDL or MYRIAD is "
+                           "acceptable. Sample will look for a suitable plugin for device specified. Default value is CPU",
+                      default="CPU", type=str)
+    args.add_argument('--preprocessing_type', choices=PREPROCESSING.keys(),
+                      help="Required. Type of the preprocessing", required=True, default='Crop')
+    args.add_argument('-pc', '--perf_counts',
+                      action='store_true', default=False)
+    return parser
+
+
+def main():
+    log.basicConfig(format="[ %(levelname)s ] %(message)s",
+                    level=log.INFO, stream=sys.stdout)
+    args = build_argparser().parse_args()
+    log.info("Creating Inference Engine")
+    ie = IECore()
+    ie.set_config(
+        {"PERF_COUNT": "YES" if args.perf_counts else "NO"}, args.device)
+
+    # encoder part
+    encoder_model_xml = args.m_encoder
+    encoder_model_bin = os.path.splitext(encoder_model_xml)[0] + ".bin"
+    log.info("Loading encoder files:\n\t{}\n\t{}".format(
+        encoder_model_xml, encoder_model_bin))
+    encoder = ie.read_network(encoder_model_xml, encoder_model_bin)
+
+    # check if all layers are supported
+    supported_layers = ie.query_network(encoder, args.device)
+    not_supported_layers = [
+        l for l in encoder.layers.keys() if l not in supported_layers]
+    if len(not_supported_layers) != 0:
+        log.error("Following layers are not supported by the plugin for specified device {}:\n{}".
+                  format(args.device, ', '.join(not_supported_layers)))
+
+    # decoder part:
+    dec_step_model_xml = args.m_decoder
+    dec_step_model_bin = os.path.splitext(dec_step_model_xml)[0] + ".bin"
+
+    log.info("Loading decoder files:\n\t{}\n\t{}".format(
+        dec_step_model_xml, dec_step_model_bin))
+    dec_step = ie.read_network(dec_step_model_xml, dec_step_model_bin)
+
+    # check if all layers are supported
+    supported_layers = ie.query_network(dec_step, args.device)
+    not_supported_layers = [
+        l for l in dec_step.layers.keys() if l not in supported_layers]
+    if len(not_supported_layers) != 0:
+        log.error("Following layers are not supported by the plugin for specified device {}:\n{}".
+                  format(args.device, ', '.join(not_supported_layers)))
+    _, _, height, width = encoder.input_info['imgs'].input_data.shape
```

Comment:
In `README.md` says, that demo works only with batch size 1. But it isn't check anywhere. Please add this check, e.g. right after this line.

Reply:
Done